engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Vector.java

```diff
@@ -111,9 +111,9 @@ public Object readArrayElement(
       @CachedLibrary(limit = "3") WarningsLibrary warnings,
       @Cached HostValueToEnsoNode toEnso)
       throws InvalidArrayIndexException, UnsupportedMessageException {
-    var v = interop.readArrayElement(storage, index);
-    if (warnings.hasWarnings(this)) {
-      Warning[] extracted = warnings.getWarnings(this, null);
+    var v = interop.readArrayElement(this.storage, index);
+    if (warnings.hasWarnings(this.storage)) {
+      Warning[] extracted = warnings.getWarnings(this.storage, null);
       if (warnings.hasWarnings(v)) {
         v = warnings.removeWarnings(v);
       }
```

Comment:
Btw. if we `removeWarnings` from `v` and then only attach `extracted`, does that mean that if `v` contained any warnings, they are discarded?

Shouldn't we rather 'merge' the two lists? Or do we guarantee that `v` will never actually contain warnings? But in that case I'd rather do `if (warnings.hasWarnings(v)) { throw impossible state error }`.

Reply:
Maybe the `warnings.getWarnings(this.storage, null)` will also extract warnings from `v`? So that it is safe to remove warnings from `v` after that?

Reply:
> Shouldn't we rather 'merge' the two lists? Or do we guarantee that v will never actually contain warnings? But in that case I'd rather do if (warnings.hasWarnings(v)) { throw impossible state error }.

That feels like just adding duplicates. Warnings of `v` are already present in warnings of `extracted`.

Reply:
Ah, ok - I did not realise that - I just wanted to double check all is good but it seems it's all fine.